src/Tools/dotnet-monitor/CollectionRules/CollectionRuleService.cs

```diff
@@ -146,12 +146,21 @@ public async ValueTask DisposeAsync()
 
                 if (null != options.Filters)
                 {
+                    IProcessInfo processInfo = await ProcessInfoImpl.FromEndpointInfoAsync(endpointInfo);
```

Comment:
This should be moved into the ApplyRules method and then pass in the IProcessInfo instance since it will be the same amongst all of the rule instances for each process.